cli/azd/pkg/pipeline/azdo_provider.go

```diff
@@ -418,56 +418,61 @@ func (p *AzdoScmProvider) promptForAzdoRepository(ctx context.Context, console i
 	return remoteUrl, nil
 }
 
-// defines the structure of an ssl git remote
-var azdoRemoteGitUrlRegex = regexp.MustCompile(`^git@ssh.dev.azure\.com:(.*?)(?:\.git)?$`)
-
-// defines the structure of an HTTPS git remote
-var azdoRemoteHttpsUrlRegex = regexp.MustCompile(`^https://[a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)*:*.+@dev.azure\.com/(.*?)$`)
-
 // ErrRemoteHostIsNotAzDo the error used when a non Azure DevOps remote is found
 var ErrRemoteHostIsNotAzDo = errors.New("existing remote is not an Azure DevOps host")
 
 // ErrSSHNotSupported the error used when ssh git remote is detected
 var ErrSSHNotSupported = errors.New("ssh git remote is not supported. " +
 	"Use HTTPS git remote to connect the remote repository")
 
-// helper function to determine if the provided remoteUrl is an azure devops repo.
-// currently supports AzDo PaaS
-func isAzDoRemote(remoteUrl string) error {
-	if azdoRemoteGitUrlRegex.MatchString(remoteUrl) {
-		return ErrSSHNotSupported
-	}
-	slug := ""
-	for _, r := range []*regexp.Regexp{azdoRemoteGitUrlRegex, azdoRemoteHttpsUrlRegex} {
-		captures := r.FindStringSubmatch(remoteUrl)
-		if captures != nil {
-			slug = captures[1]
-		}
-	}
-	if slug == "" {
-		return ErrRemoteHostIsNotAzDo
-	}
-	return nil
+type azdoRemote struct {
+	Project        string
+	RepositoryName string
 }
 
-func parseAzDoRemote(remoteUrl string) (string, error) {
-	for _, r := range []*regexp.Regexp{azdoRemoteGitUrlRegex, azdoRemoteHttpsUrlRegex} {
-		captures := r.FindStringSubmatch(remoteUrl)
-		if captures != nil {
-			return captures[1], nil
-		}
+var httpsRegex = regexp.MustCompile(
+	//nolint:lll
+	`^https:\/\/(?:.*)?(dev\.azure\.com|visualstudio\.com)\/([a-zA-Z0-9]+)\/([a-zA-Z0-9]+)\/_git\/([a-zA-Z0-9]+)$`)
+
+var sshRegex = regexp.MustCompile(
+	//nolint:lll
+	`^git@(?:ssh\.dev\.azure\.com|vs-ssh\.visualstudio\.com|ssh\.visualstudio\.com):(v[1-3])?\/([a-zA-Z0-9]+)\/([a-zA-Z0-9]+)\/([a-zA-Z0-9]+)$`)
+
+// parseAzDoRemote extracts the organization, project and repository name from an Azure DevOps remote url
+// the url can be in the form of:
+//   - https://dev.azure.com/[org|user]/[project]/_git/[repo]
+//   - https://[user]@dev.azure.com/[org|user]/[project]/_git/[repo]
+//   - https://[org].visualstudio.com/[project]/_git/[repo]
+//   - git@ssh.dev.azure.com:v[1-3]/[user|org]/[project]/[repo]
+//   - git@vs-ssh.visualstudio.com:v[1-3]/[user|org]/[project]/[repo]
```

Comment:
In the AzDo UI for the older formats, they give a SCP-like [git remote url](https://git-scm.com/docs/git-clone#_git_urls) with `git@` omitted. It looks like:

`sanitizedOrg@vs-ssh.visualstudio.com:v3/sanitizedOrg/sanitizedProject/sanitizedRepo`